### DISCUSSION TITLE
how to erase the voltage low level between each 8-bit in SCK(SPI)

### DISCUSSION THREAD

**Discussion by carl336** [micropython/micropython#14192] [RP2040 / Pico]
Hi, I'm a starter and would like to ask a question about SPI SCK.

Hardware→Raspberry Pi Pico (RP2040, MicroPython v1.22.2 )
I am using Pico to turn a BL on with SPI, and the code writes with Micropython.  

The point is :
I want to let SCK send a continuous and uninterrupted signal, but I don't know how to deal with it. 
（I uploaded a picture of a signal detected with a logic analyzer, it shows that there are some voltage low levels of signal between each 8 bits, but I want to erase them.）
Could anybody teach me?
Here is the code that I wrote: 
https://paste.fastmirror.net/?b5d8eb2f87a857da#GMJkrrTLhkVuXJvWzEPpDmnzp61MhCb3z3n4QroXDYAr 

And here is the screenshot of a signal detected with a logic analyzer (Kingst VIS).
https://fastupload.io/71tuEoAbk1jQZDd/file

The purpose of having SCK send signals continuously and uninterruptedly is that there is a Backlight LED Driver, according to its 
specification, it looks like I need to keep the signals from SCK  continuously and uninterruptedly.
https://fastupload.io/YUs89PQEr1j3xby/file

**Comment by davefes** [comment]
File has been deleted.

**Reply by carl336** [reply]
Thanks for noticing me, I just reedited it.

**Reply by davefes** [reply]
I can only suggest that you investigate whether or not pull-down resistors are needed or to configure the SPI pins as `pull-down`.  You might need an expert on this issue.

**Reply by robert-hh** [reply]
With SPI, the Pin for both SCK and MOSI are configure for actively driving the high and low level. So no pull-up/down resistors are needed. If you still see some signals at the SCK line when it should be low, check your wiring. Is GND attached? Are the wires and connections good? Are there short to other pins? Is the line driven by a different signal as well?

Edit: b.t.w. The above links for the diagrams are still invalid. You have to edit your post and insert the new links.
Edit 2: You can directly paste pictures or screen shots into your post. No need to link an external file storage. Just pull them from the file manager into the text box of your post.

**Reply by carl336** [reply]
thanks for your advice, I just renewed those two links.

And please let me answer your questions:
Is GND attached? →Yes.
Are the wires and connections good?  →Yes, all good.
Are there short to other pins? →No, everything is fine.
 Is the line driven by a different signal as well? →No,each line is only driven by one signal.

**Reply by robert-hh** [reply]
OK. So as a SPI interface, the signals are fine and look as they should. The simple fact is, that you cannot get with SPI what you want to obtain. But I see that the level of the MOSI signal is low, when not used. Maybe you should change that to high at the first step. I ordered the detailed data sheet of the device, which may take some time to arrive.
